mlrun/api/utils/clients/chief.py

```diff
@@ -0,0 +1,127 @@
+import asyncio
+import copy
+
+import fastapi
+import requests.adapters
+import urllib3
+
+import mlrun.api.schemas
+import mlrun.api.utils.projects.remotes.follower
+import mlrun.errors
+import mlrun.utils.singleton
+from mlrun.utils import logger
+
+
+# we were thinking to simply use httpdb, but decided to have a separated class for simplicity for now until
+# this class evolves, but this should be reconsidered when adding more functionality to the class
+class Client(
+    metaclass=mlrun.utils.singleton.AbstractSingleton,
+):
+    """
+    We chose chief-workers architecture to provide multi-instance API.
+    By default, all API calls can access both the chief and workers.
+    The key distinction is that some responsibilities, such as scheduling jobs, are exclusively performed by the chief.
+    Instead of limiting the ui/client to only send requests to the chief, because the workers doesn't hold all the
+    information. When one of the workers receives a request that the chief needs to execute or may have the knowledge
+    of that piece of information, the worker will redirect the request to the chief.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        http_adapter = requests.adapters.HTTPAdapter(
+            max_retries=urllib3.util.retry.Retry(total=3, backoff_factor=1),
+            pool_maxsize=int(mlrun.mlconf.httpdb.max_workers),
+        )
+        self._session = requests.Session()
+        self._session.mount("http://", http_adapter)
+        self._api_url = mlrun.mlconf.resolve_chief_api_url()
+
+    def get_background_task(
+        self, name: str, request: fastapi.Request = None
+    ) -> fastapi.Response:
+        request_kwargs = self._resolve_request_kwargs_from_request(request)
+        chief_response = self._send_request_to_api(
+            "GET", f"background-tasks/{name}", **request_kwargs
+        )
+        return self._parse_response_to_fastapi_response(chief_response)
```

Comment:
It seems redundant that you're calling `_send_request_to_api` and then `_parse_response_to_fastapi_response` again and again
Maybe add a `_proxy_request_to_chief` that will call those two